File: guests/swift/a11yrows.swift
// The a11yrows scene, Swift port — guests/rust/a11yrows.rs,
// tools/scenes/a11yrows.steps.

import Foundation

let app = KayaApp()

app.build { tx in
    let notes = tx.collection()
    let heads = tx.collection()
    let root = tx.column {
        tx.each(notes) { t in
            // Element-sourced: expect_ax refuses an ambiguous authored id
            // (docs/tpl-props-plan.md).
            let field = t.entry()  // entry#0 and entry#1 once the rows stamp
            t.setA11yId(field, KayaField<String>.element)
            t.setA11yLabel(field, KayaField<String>.element)
        }

        // A SECOND COLLECTION because a scalar row has one field to spend
        // on an id.
        tx.each(heads) { t in
            let bar = t.row {
                let title = t.label(KayaField<String>.element)
                t.setRole(title, .heading)
                t.setA11yId(title, KayaField<String>.element)
            }
            t.setInset(bar, 8)
        }
    }
    tx.insertFresh(notes, .str("First note"))
    tx.insertFresh(notes, .str("Second note"))
    tx.insertFresh(heads, .str("Heading one"))
    tx.insertFresh(heads, .str("Heading two"))
    tx.mount(root)
}

app.run()

File: guests/ocaml/a11yrows.ml
(* The a11yrows scene, OCaml port — guests/rust/a11yrows.rs,
   tools/scenes/a11yrows.steps. *)

open Kaya_wire
open Kaya_app

let () =
  let app = Kaya_app.create () in

  build app (fun () ->
     let notes = collection () in
     let heads = collection () in

     let root =
       column
         [
           (* [~a11y_id_field], not [~a11y_id]: [expect_ax] refuses an
              ambiguous authored identifier (docs/deferred.md). *)
           each notes (fun () ->
              Tpl.(entry ~a11y_id_field:element ~a11y_label_field:element ()));
           (* A SECOND collection because a scalar row has one field to
              spend on an id. *)
           each heads (fun () ->
              Tpl.(
                row ~inset:8.0
                  [ label ~role:Heading ~bind_field:element ~a11y_id_field:element ]
                  ()));
         ]
         ()
     in
     mount root;
     ignore (insert_fresh notes (Str "First note"));
     ignore (insert_fresh notes (Str "Second note"));
     ignore (insert_fresh heads (Str "Heading one"));
     ignore (insert_fresh heads (Str "Heading two")));

  exit (run app)
